azkaban-web-server/src/main/java/azkaban/imagemgmt/services/ImageVersionServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package azkaban.imagemgmt.services;
+
+import azkaban.imagemgmt.daos.ImageVersionDao;
+import azkaban.imagemgmt.exeception.ImageMgmtException;
+import azkaban.imagemgmt.exeception.ImageMgmtValidationException;
+import azkaban.imagemgmt.models.ImageVersion;
+import azkaban.imagemgmt.models.ImageVersion.State;
+import azkaban.imagemgmt.models.RequestContext;
+import azkaban.imagemgmt.utils.ValidatorUtils;
+import java.io.IOException;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.codehaus.jackson.map.ObjectMapper;
+
+@Singleton
+public class ImageVersionServiceImpl implements ImageVersionService {
+
+  private final ImageVersionDao imageVersionsDao;
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  public ImageVersionServiceImpl(final ImageVersionDao imageVersionsDao, final ObjectMapper objectMapper) {
+    this.imageVersionsDao = imageVersionsDao;
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public int createImageVersion(RequestContext requestContext)
+      throws IOException, ImageMgmtException {
+    ImageVersion imageVersion = objectMapper.readValue(requestContext.getJsonPayload(), ImageVersion.class);
+    imageVersion.setCreatedBy(requestContext.getUser());
+    if(imageVersion.getState() == null) {
+      imageVersion.setState(State.NEW);
+    }
+    if(!ValidatorUtils.validateObject(imageVersion)) {
+      throw new ImageMgmtValidationException("Provide valid input for creating image version "
+          + "metadata");
+    }
+    return imageVersionsDao.createImageVersion(imageVersion);
+  }
+
+  @Override
+  public List<ImageVersion> getImageVersion(RequestContext requestContext) throws ImageMgmtException {
+    return imageVersionsDao.getImageVersion(requestContext);
```

Comment:
RequestContext should not be passed to the DAO.

Reply:
RequestContext is a data transfer object and it just passes the data across layers. Internal builder is provided and hence the request builder can be build with required parameter. With this context, if we pass RequestContext to DAO methods specially get/fetch etc. we need to pass multiple parameters. We can add different parameter in the RequestContext. Moreover, pagination parameters can also be passed across. This is specially useful for any search, get type methods.